chatterbot/storage/mongodb.py

```diff
@@ -331,27 +381,43 @@ def remove(self, statement_text):
 
         self.statements.delete_one({'text': statement_text})
 
-    def get_response_statements(self):
+    def get_response_statements(self, statement=None):
         """
         Return only statements that are in response to another statement.
         A statement must exist which lists the closest matching statement in the
         in_response_to field. Otherwise, the logic adapter may find a closest
         matching statement that does not have a known response.
         """
-        response_query = self.statements.aggregate([{'$group': {'_id': '$in_response_to.text'}}])
-
-        responses = []
-        for r in response_query:
-            try:
-                responses.extend(r['_id'])
-            except TypeError:
-                pass
-
-        _statement_query = {
-            'text': {
-                '$in': responses
+        if not statement:
+            response_query = self.statements.aggregate(
+                [{'$group': {'_id': '$in_response_to.text'}}])
+
+            responses = []
+            for r in response_query:
+                try:
+                    responses.extend(r['_id'])
+                except TypeError:
+                    pass
+
+            _statement_query = {
+                'text': {
+                    '$in': responses
+                }
+            }
+        else:
+            # Just filter the statement in need.
+            statement_ids = []
+            tokens = self.statement_segmentation(statement.text)
+            word_dicts = self.inverted.find({'word': {'$in': tokens}})
```

Comment:
I might be misunderstanding this, so feel free to correct me if I'm wrong. Given this change, wont only statements in which the words of `statement` exist be returned?

Reply:
Yes, you're right. So this change will lead to no response if there's no statement exist.